### DISCUSSION TITLE
Support nested directories for migrations

### DISCUSSION THREAD

**Discussion by Nutomic** [diesel-rs/diesel#3731] [Ideas]
Our [migrations folder](https://github.com/LemmyNet/lemmy/tree/main/migrations) has grown to almost 200 items over the years. It is getting harder and harder to browse, especially as you are usually only interested in the latest migrations. It would be very helpful if diesel could support nested folders like this:
```/
migrations/
├── 00000000000000_diesel_initial_setup
├── 2019
│   ├── 2019-02-26-002946_create_user
│   ├── 2019-02-27-170003_create_community
│   ├── 2019-03-03-163336_create_post
│   ├── ...
├── 2020
│   ├── 2020-01-01-200418_add_email_to_user_view
│   ├── 2020-01-02-172755_add_show_avatar_and_email_notifications_to_user
│   ├── 2020-01-11-012452_add_indexes
│   ├── ...
├── ...
```

**Comment by weiznich** [comment]
Thanks for writing this proposal. I'm not opposed to such a change, which means I'm happy to accept contributions that add this feature as long as:

* It's done backward compatible in a strictly optional way
* It's covered by tests
